modules/http/src/main/java/org/mule/module/http/internal/listener/HttpResponseBuilder.java

```diff
@@ -208,11 +221,24 @@ else if (payload instanceof InputStream)
         return httpResponseBuilder.build();
     }
 
-    private void resolveEncoding(HttpResponseHeaderBuilder httpResponseHeaderBuilder, String existingTransferEncoding, String existingContentLength, ByteArrayHttpEntity byteArrayHttpEntity)
+    private boolean supportsTransferEncoding(MuleEvent event)
+    {
+        String httpVersion = event.getMessage().<String> getInboundProperty(HTTP_VERSION_PROPERTY);
+        return !(Protocol.HTTP_0_9.getProtocolString().equals(httpVersion) || Protocol.HTTP_1_0.getProtocolString().equals(httpVersion));
```

Comment:
Use org.mule.module.http.internal.domain.HttpProtocol enum instead of a class from Grizzly (if necessary, add 0.9 version to this enum)